inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/metrics/MeasuresAndViewsManager.java

```diff
@@ -0,0 +1,298 @@
+                new ArrayList<>(viewTags));
+        viewManager.registerView(view);
+    }
+
+    private void updateView(ViewDefinitionSettings def, Set<TagKey> viewTags, View view) {
+        if (!def.getDescription().equals(view.getDescription())) {
+            log.info("Cannot update description of view '{}' because it has been already registered in OpenCensus!", def.getName());
+        }
+        if (!isAggregationOfType(def.getAggregation(), view.getAggregation())) {
+            log.info("Cannot update aggregation of view '{}' because it has been already registered in OpenCensus!", def.getName());
+        }
+        Set<TagKey> presentTagKeys = new HashSet<>(view.getColumns());
+
+        presentTagKeys.stream()
+                .filter(t -> !viewTags.contains(t))
+                .forEach(tag -> log.info("Cannot remove tag '{}' from view '{}' because it has been already registered in OpenCensus!", tag.getName(), def.getName()));
+        viewTags.stream()
+                .filter(t -> !presentTagKeys.contains(t))
+                .forEach(tag -> log.info("Cannot add tag '{}' to view '{}' because it has been already registered in OpenCensus!", tag.getName(), def.getName()));
+    }
+
+    /**
+     * Collects the tags which should be used for the given view.
+     * This function includes the common tags if requested,
+     * then applies the {@link ViewDefinitionSettings#getTags()}
+     * and finally converts them to {@link TagKey}s.
+     *
+     * @param def the view definition for which the tags should be collected.
+     * @return the set of tag keys
+     */
+    private Set<TagKey> getTagKeysForView(ViewDefinitionSettings def) {
+        Set<TagKey> viewTags = new HashSet<>();
+        if (def.isWithCommonTags()) {
+            commonTags.getCommonTagKeys().stream()
+                    .filter(t -> def.getTags().get(t.getName()) != Boolean.FALSE)
+                    .forEach(viewTags::add);
+        }
+        def.getTags().entrySet().stream()
+                .filter(Map.Entry::getValue)
+                .map(Map.Entry::getKey)
+                .map(TagKey::create)
+                .forEach(viewTags::add);
+        return viewTags;
+    }
+
+    private boolean isAggregationOfType(ViewDefinitionSettings.Aggregation aggregationType, Aggregation instance) {
+        switch (aggregationType) {
+            case COUNT:
+                return instance instanceof Aggregation.Count;
+            case SUM:
+                return instance instanceof Aggregation.Sum;
+            case HISTOGRAM:
+                return instance instanceof Aggregation.Distribution;
+            case LAST_VALUE:
+                return instance instanceof Aggregation.LastValue;
+            default:
+                throw new RuntimeException("Unhandled case!");
+        }
+    }
+
+
+    private Aggregation getAggregationOfType(ViewDefinitionSettings.Aggregation aggregationType, List<Double> histogramBuckets) {
```

Comment:
Optional Comment:
how about changing the signatare to `getViewAggregation(ViewDefinitionSettings definitionSettings)`?